Mypy passes locally for me, as `typing_test_data.py` is excluded, but fails on CI, not sure what's going on here:

```
tests/test_util/test_util_inspect.py:251: note: In module imported here:
tests/test_util/typing_test_data.py: note: In function "f1":
tests/test_util/typing_test_data.py:10:1: error: Missing return statement  [empty-body]
tests/test_util/typing_test_data.py: note: In function "f2":
tests/test_util/typing_test_data.py:19:1: error: Missing return statement  [empty-body]
tests/test_util/typing_test_data.py: note: In function "f6":
tests/test_util/typing_test_data.py:38:1: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f7":
tests/test_util/typing_test_data.py:42:17: error: Incompatible default for argument "x" (default has type "None", argument has type "int")  [assignment]
tests/test_util/typing_test_data.py:42:17: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tests/test_util/typing_test_data.py:42:17: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tests/test_util/typing_test_data.py:42:26: error: Missing type parameters for generic type "dict"  [type-arg]
tests/test_util/typing_test_data.py: note: In function "f9":
tests/test_util/typing_test_data.py:51:11: error: Missing type parameters for generic type "Callable"  [type-arg]
tests/test_util/typing_test_data.py: note: In member "__repr__" of class "CustomAnnotation":
tests/test_util/typing_test_data.py:60:5: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f11":
tests/test_util/typing_test_data.py:64:12: error: Invalid type comment or annotation  [valid-type]
tests/test_util/typing_test_data.py:64:12: note: Suggestion: use CustomAnnotation[...] instead of CustomAnnotation(...)
tests/test_util/typing_test_data.py:64:35: error: Invalid type: try using Literal[123] instead?  [valid-type]
tests/test_util/typing_test_data.py: note: In function "f12":
tests/test_util/typing_test_data.py:68:1: error: Missing return statement  [empty-body]
tests/test_util/typing_test_data.py: note: In function "f15":
tests/test_util/typing_test_data.py:80:12: error: Name "Unknown" is not defined  [name-defined]
tests/test_util/typing_test_data.py: note: In function "f16":
tests/test_util/typing_test_data.py:84:1: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f17":
tests/test_util/typing_test_data.py:88:1: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f18":
tests/test_util/typing_test_data.py:92:1: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tests/test_util/typing_test_data.py:92:1: error: Missing return statement  [empty-body]
tests/test_util/typing_test_data.py:92:32: error: Missing type parameters for generic type "Tuple"  [type-arg]
tests/test_util/typing_test_data.py:92:53: error: Missing type parameters for generic type "Dict"  [type-arg]
tests/test_util/typing_test_data.py: note: In function "f19":
tests/test_util/typing_test_data.py:96:1: error: Function is missing a return type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f21":
tests/test_util/typing_test_data.py:104:1: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f22":
tests/test_util/typing_test_data.py:108:1: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f23":
tests/test_util/typing_test_data.py:112:1: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f24":
tests/test_util/typing_test_data.py:116:1: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In function "f25":
tests/test_util/typing_test_data.py:120:1: error: Function is missing a type annotation  [no-untyped-def]
tests/test_util/typing_test_data.py: note: In member "children" of class "Node":
tests/test_util/typing_test_data.py:132:5: error: Missing return statement  [empty-body]
Found 24 errors in 1 file (checked 379 source files)
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -197,9 +197,7 @@ module = [
     "tests.test_events",
     "tests.test_highlighting",
     "tests.test_project",
-    "tests.test_quickstart",
     "tests.test_roles",
-    "tests.test_search",
     "tests.test_versioning",
     # tests/test__cli
     "tests.test__cli.test__cli_util_errors",
@@ -207,8 +205,6 @@ module = [
     "tests.test_builders.test_build",
     "tests.test_builders.test_build_changes",
     "tests.test_builders.test_build_dirhtml",
-    "tests.test_builders.test_build_epub",
-    "tests.test_builders.test_build_gettext",
     "tests.test_builders.test_build_html",
     "tests.test_builders.test_build_html_5_output",
     "tests.test_builders.test_build_html_assets",
@@ -221,43 +217,25 @@ module = [
     "tests.test_builders.test_build_html_numfig",
     "tests.test_builders.test_build_html_tocdepth",
     "tests.test_builders.test_build_html_toctree",
-    "tests.test_builders.test_build_latex",
     "tests.test_builders.test_build_linkcheck",
     "tests.test_builders.test_build_manpage",
-    "tests.test_builders.test_build_texinfo",
     "tests.test_builders.test_build_text",
     "tests.test_builders.test_build_warnings",
     "tests.test_builders.test_incremental_reading",
     # tests/test_config
-    "tests.test_config.test_config",
     "tests.test_config.test_copyright",
     # tests/test_directives
     "tests.test_directives.test_directive_code",
     "tests.test_directives.test_directive_object_description",
-    "tests.test_directives.test_directive_only",
     "tests.test_directives.test_directive_option",
-    "tests.test_directives.test_directive_other",
-    "tests.test_directives.test_directive_patch",
     "tests.test_directives.test_directives_no_typesetting",
     # tests/test_domains
-    "tests.test_domains.test_domain_c",
-    "tests.test_domains.test_domain_cpp",
-    "tests.test_domains.test_domain_js",
-    "tests.test_domains.test_domain_py",
     "tests.test_domains.test_domain_py_canonical",
-    "tests.test_domains.test_domain_py_fields",
-    "tests.test_domains.test_domain_py_pyfunction",
-    "tests.test_domains.test_domain_py_pyobject",
-    "tests.test_domains.test_domain_rst",
-    "tests.test_domains.test_domain_std",
     # tests/test_environment
     "tests.test_environment.test_environment",
     "tests.test_environment.test_environment_indexentries",
     "tests.test_environment.test_environment_record_dependencies",
-    "tests.test_environment.test_environment_toctree",
     # tests/test_extensions
-    "tests.test_extensions.test_ext_apidoc",
-    "tests.test_extensions.test_ext_autodoc",
     "tests.test_extensions.test_ext_autodoc_autoattribute",
     "tests.test_extensions.test_ext_autodoc_autoclass",
     "tests.test_extensions.test_ext_autodoc_autodata",
@@ -265,15 +243,11 @@ module = [
     "tests.test_extensions.test_ext_autodoc_automodule",
     "tests.test_extensions.test_ext_autodoc_autoproperty",
     "tests.test_extensions.test_ext_autodoc_configs",
-    "tests.test_extensions.test_ext_autodoc_events",
-    "tests.test_extensions.test_ext_autodoc_mock",
     "tests.test_extensions.test_ext_autodoc_preserve_defaults",
     "tests.test_extensions.test_ext_autodoc_private_members",
     "tests.test_extensions.test_ext_autosectionlabel",
-    "tests.test_extensions.test_ext_autosummary",
     "tests.test_extensions.test_ext_autosummary_imports",
     "tests.test_extensions.test_ext_coverage",
-    "tests.test_extensions.test_ext_doctest",
     "tests.test_extensions.test_ext_duration",
     "tests.test_extensions.test_ext_extlinks",
     "tests.test_extensions.test_ext_githubpages",
@@ -281,18 +255,14 @@ module = [
     "tests.test_extensions.test_ext_ifconfig",
     "tests.test_extensions.test_ext_imgconverter",
     "tests.test_extensions.test_ext_imgmockconverter",
-    "tests.test_extensions.test_ext_inheritance_diagram",
-    "tests.test_extensions.test_ext_intersphinx",
     "tests.test_extensions.test_ext_intersphinx_cache",
     "tests.test_extensions.test_ext_math",
     "tests.test_extensions.test_ext_napoleon",
-    "tests.test_extensions.test_ext_napoleon_docstring",
     "tests.test_extensions.test_ext_todo",
     "tests.test_extensions.test_ext_viewcode",
     "tests.test_extensions.test_extension",
     # tests/test_intl
     "tests.test_intl.test_catalogs",
-    "tests.test_intl.test_intl",
     "tests.test_intl.test_locale",
     # tests/test_markup
     "tests.test_markup.test_markup",
@@ -300,8 +270,6 @@ module = [
     "tests.test_markup.test_parser",
     "tests.test_markup.test_smartquotes",
     # tests/test_pycode
-    "tests.test_pycode.test_pycode",
-    "tests.test_pycode.test_pycode_ast",
     "tests.test_pycode.test_pycode_parser",
     # tests/test_theming
     "tests.test_theming.test_html_theme",
@@ -309,7 +277,6 @@ module = [
     "tests.test_theming.test_theming",
     # tests/test_transforms
     "tests.test_transforms.test_transforms_move_module_targets",
-    "tests.test_transforms.test_transforms_post_transforms",
     "tests.test_transforms.test_transforms_post_transforms_code",
     "tests.test_transforms.test_transforms_post_transforms_images",
     "tests.test_transforms.test_transforms_reorder_nodes",
@@ -319,24 +286,73 @@ module = [
     "tests.test_util.test_util_docstrings",
     "tests.test_util.test_util_docutils",
     "tests.test_util.test_util_docutils_sphinx_directive",
-    "tests.test_util.test_util_fileutil",
-    "tests.test_util.test_util_i18n",
     "tests.test_util.test_util_images",
     "tests.test_util.test_util_importer",
-    "tests.test_util.test_util_inspect",
     "tests.test_util.test_util_inventory",
     "tests.test_util.test_util_lines",
-    "tests.test_util.test_util_logging",
     "tests.test_util.test_util_matching",
-    "tests.test_util.test_util_nodes",
-    "tests.test_util.test_util_rst",
-    "tests.test_util.test_util_template",
-    "tests.test_util.test_util_typing",
     "tests.test_util.test_util_uri",
     # tests/test_writers
     "tests.test_writers.test_api_translator",
     "tests.test_writers.test_docutilsconf",
     "tests.test_writers.test_writer_latex",
+]
+disallow_untyped_defs = false
+
+[[tool.mypy.overrides]]
+module = [
+    # tests/
+    "tests.test_quickstart",
+    "tests.test_search",
+    # tests/test_builders
+    "tests.test_builders.test_build_epub",
+    "tests.test_builders.test_build_gettext",
+    "tests.test_builders.test_build_latex",
+    "tests.test_builders.test_build_texinfo",
+    # tests/test_config
+    "tests.test_config.test_config",
+    # tests/test_directives
+    "tests.test_directives.test_directive_only",
+    "tests.test_directives.test_directive_other",
+    "tests.test_directives.test_directive_patch",
+    # tests/test_domains
+    "tests.test_domains.test_domain_c",
+    "tests.test_domains.test_domain_cpp",
+    "tests.test_domains.test_domain_js",
+    "tests.test_domains.test_domain_py",
+    "tests.test_domains.test_domain_py_fields",
+    "tests.test_domains.test_domain_py_pyfunction",
+    "tests.test_domains.test_domain_py_pyobject",
+    "tests.test_domains.test_domain_rst",
+    "tests.test_domains.test_domain_std",
+    # tests/test_environment
+    "tests.test_environment.test_environment_toctree",
+    # tests/test_extensions
+    "tests.test_extensions.test_ext_apidoc",
+    "tests.test_extensions.test_ext_autodoc",
+    "tests.test_extensions.test_ext_autodoc_events",
+    "tests.test_extensions.test_ext_autodoc_mock",
+    "tests.test_extensions.test_ext_autosummary",
+    "tests.test_extensions.test_ext_doctest",
+    "tests.test_extensions.test_ext_inheritance_diagram",
+    "tests.test_extensions.test_ext_intersphinx",
+    "tests.test_extensions.test_ext_napoleon_docstring",
+    # tests/test_intl
+    "tests.test_intl.test_intl",
+    # tests/test_pycode
+    "tests.test_pycode.test_pycode",
+    "tests.test_pycode.test_pycode_ast",
+    # tests/test_transforms
+    "tests.test_transforms.test_transforms_post_transforms",
+    # tests/test_util
+    "tests.test_util.test_util_fileutil",
+    "tests.test_util.test_util_i18n",
+    "tests.test_util.test_util_inspect",
+    "tests.test_util.test_util_logging",
+    "tests.test_util.test_util_nodes",
+    "tests.test_util.test_util_rst",
+    "tests.test_util.test_util_template",
+    "tests.test_util.test_util_typing",
 ]
 check_untyped_defs = false
 disable_error_code = [
@@ -345,7 +361,6 @@ disable_error_code = [
 disallow_incomplete_defs = false
 disallow_untyped_calls = false
 disallow_untyped_defs = false
-warn_unused_ignores = false
 
 [[tool.mypy.overrides]]
 module = ["tests.test_util.typing_test_data"]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -146,8 +146,6 @@ files = [
 ]
 exclude = [
     "tests/roots",
-    # tests/test_util
-    "^tests/test_util/typing_test_data\\.py$",
 ]
 python_version = "3.11"
 strict = true
@@ -348,6 +346,11 @@ disallow_incomplete_defs = false
 disallow_untyped_calls = false
 disallow_untyped_defs = false
 warn_unused_ignores = false
+
+[[tool.mypy.overrides]]
+
+module = ["tests.test_util.typing_test_data"]
+ignore_errors = true
 
 [tool.pytest.ini_options]
 minversion = "6.0"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -348,7 +348,6 @@ disallow_untyped_defs = false
 warn_unused_ignores = false
 
 [[tool.mypy.overrides]]
-
 module = ["tests.test_util.typing_test_data"]
 ignore_errors = true
 

--- a/tests/test_extensions/test_ext_autodoc_events.py
+++ b/tests/test_extensions/test_ext_autodoc_events.py
@@ -71,7 +71,7 @@ def test_cut_lines_no_objtype():
     ]
     process = cut_lines(2)
 
-    process(None, 'function', 'func', None, {}, docstring_lines)  # type: ignore[arg-type]
+    process(None, 'function', 'func', None, {}, docstring_lines)
     assert docstring_lines == [
         'second line',
         '---',

